dask/array/linalg.py

```diff
@@ -328,3 +328,117 @@ def svd(a, name=None):
+        if i == 0:
+            dsk[(name_lu, i, i)] = (scipy.linalg.lu, (a.name, i, i))
+            for j in range(i + 1, hdim):
+                # sweep to horizontal
+                dsk[(name_lu, i, j)] = (_lu_horizontal, (a.name, i, j), (name_l_internal, i, i))
+            for k in range(i + 1, vdim):
+                # sweep to vertical
+                dsk[(name_lu, k, i)] = (_lu_vertical, (a.name, k, i), (name_u_internal, i, i))
+        else:
+            dsk[(name_lu, i, i)] = (_lu_diag, (a.name, i, i), (name_l_internal, i, i - 1),
+                                    (name_u_internal, i - 1, i))
+
+            for j in range(i + 1, hdim):
+                # sweep to horizontal
+                dsk[(name_lu, i, j)] = (_lu_horizontal_diag, (a.name, i, j),
+                                                             (name_l_internal, i, 1),
+                                                             (name_u_internal, i - 1, j),
+                                                             (name_l_internal, i, i))
+            for k in range(i + 1, vdim):
+                # sweep to vertical
+                dsk[(name_lu, k, i)] = (_lu_vertical_diag, (a.name, k, i),
+                                                           (name_l_internal, k, 1),
+                                                           (name_u_internal, 1, i)
+                                                           (name_u_internal, i, i))
+
+        for j in range(i + 1, hdim):
+            # sweep to horizontal
+            dsk[(name_lu, i, j)] = (_lu_horizontal, (a.name, i, j), (name_l_internal, i, i))
+        for k in range(i + 1, vdim):
+            # sweep to vertical
+            dsk[(name_lu, k, i)] = (_lu_vertical, (a.name, k, i), (name_u_internal, i, i))
+
+    for i in range(min(vdim, hdim)):
+        for j in range(min(vdim, hdim)):
+            if i == j:
+                dsk[(name_l_internal, i, j)] = (operator.getitem, (name_lu, i, j), 1)
+                dsk[(name_u_internal, i, j)] = (operator.getitem, (name_lu, i, j), 2)
+
+                dsk[(name_p, i, j)] = (operator.getitem, (name_lu, i, j), 0)
+
+            elif i > j:
+                # ToDo: remove _dummy_identity
+                dsk[(name_p, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_l_internal, i, j)] = (_dummy_identity, (name_lu, i, j))
+                dsk[(name_u_internal, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+            else:
+                dsk[(name_p, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_l_internal, i, j)] = (np.zeros, (a.chunks[0][i], a.chunks[1][j]))
+                dsk[(name_u_internal, i, j)] = (_dummy_identity, (name_lu, i, j))
+
+            dsk[(name_l, i, j)] = (_dummy_identity, (name_l_internal, i, j))
+            dsk[(name_u, i, j)] = (_dummy_identity, (name_u_internal, i, j))
+
+    dsk.update(a.dask)
+
+    dsk_p = dsk.copy()
```

Comment:
Is this necessary? @mrocklin any thoughts on if it's dangerous to share a task dict between multiple arrays?